### PR TITLE
Change SIGKILL to SIGHUP

### PR DESCRIPTION
**Pull request by ash-m**
Fix for:
> Fatal error: Error installing signal handler for 9 in /app/vendor/react/event-loop/src/StreamSelectLoop.php on line 161

From https://www.php.net/manual/en/function.pcntl-signal.php#83981 (14 years old):
>  You cannot assign a signal handler for SIGKILL (kill -9).

### CODE REVIEW

File: src/reactphp/src/ServerFactory.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ public function createServer(RequestHandlerInterface $requestHandler): LoopInter
         $loop->addSignal(SIGTERM, function (int $signal) {
             exit(128 + $signal);
         });
-        $loop->addSignal(SIGKILL, function (int $signal) {
+        $loop->addSignal(SIGHUP, function (int $signal) {
```
Comment (alexander-schranz) [inline src/reactphp/src/ServerFactory.php:39]: I'm sadly not enough familiar here with this kind of signals. @Nyholm maybe you can have a look at this.

I did just stumble over the examples where the signal is also removed: https://github.com/reactphp/event-loop/blob/21914ff1db4260ca4c81e338c7e235a4df4ef94b/examples/04-signals.php#L14 is that also simething we are required but maybe not required as we are exiting here.

Comment (ash-m) [reply src/reactphp/src/ServerFactory.php:39]: yeah, I thought it might be removable entirely.

**Comment by ash-m** [comment]
any idea on if this can be merged?

**Review by alexander-schranz — APPROVED** [review]

**Comment by alexander-schranz** [comment]
Looks fine from my side as this was implemented by @brettmc. Maybe he can also confirm this changes be fine.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]
This looks wrong. sigterm is already stronger than sighup. the point of the sigkill is to send a stronger signal than the sigterm if the sigterm didn't work.

**Comment by andrew-demb** [comment]
> This looks wrong. sigterm is already stronger than sighup. the point of the sigkill is to send a stronger signal than the sigterm if the sigterm didn't work.

Please check https://www.gnu.org/software/libc/manual/html_node/Termination-Signals.html

Seems like handling SIGKILL is impossible from PHP

**Comment by GrahamCampbell** [comment]
That is intentional. The sigkill does not allow handling, and is only sent if a gracefull stop does not suceed.

**Comment by brettmc** [comment]
I don't think changing it to `SIGHUP` is correct, as that's not (usually) a terminating signal (I've often seen it used as a "restart" or "reload config" signal).
But I think it's probably correct that `SIGKILL` is not worth catching, since the process should be terminated in such a way that the application doesn't get a say in it. That's certainly true in docker/k8s, but I wasn't sure that it was _always_ true.
If `SIGKILL` is not compatible with `StreamSelectLoop` (I only tested against libev), then I don't see any harm in removing entirely.

**Review by alexander-schranz — COMMENTED** [review]
Based on the comments I would would just remove the SIGKILL part and not add SIGHUP.

**Comment by alexander-schranz** [comment]
just removing currently the SIGKILL. Thx all for the feedback and  @ash-m for the PR. If somebody has other suggestion let me know and feel free to create a pull request.

**Pull request merged**